### ISSUE TITLE
W1b: `RandomCrop` with `pad_if_needed=False` silently upscales a crop larger than the input

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4414]
`RandomCrop(size)` with a `size` larger than the input and `pad_if_needed=False` neither raises nor pads. It builds a crop box spanning the requested size — whose indices run past the end of the input — and hands it to `crop_by_indices`, which resamples. The result is an interpolated upscale of the whole input, returned under a name that promises a crop.

The class already has the branch that handles this case: `pad_if_needed=True` pads symmetrically to reach the requested size and reports the padding in `_params["padding_size"]`. With the flag left at its default the same configuration produces a silently different kind of output, and nothing in the parameters distinguishes the two — `_params["src"]` simply contains out-of-range corners.

The practical consequence is a resolution change that looks like a crop. A pipeline that crops to a fixed patch size and meets one undersized image in a mixed-size dataset gets an upscaled image rather than an error, and any downstream size or scale assumption is silently wrong for that sample.

## Repro

```python
import torch, kornia, kornia.augmentation as K
print("kornia", kornia.__version__, "torch", torch.__version__)
torch.manual_seed(0)
x = torch.arange(48., dtype=torch.float32).reshape(1, 1, 6, 8) / 48.   # 6 x 8
aug = K.RandomCrop((10, 10), pad_if_needed=False, p=1.0)
o = aug(x)
print("input", tuple(x.shape), "-> requested crop (10, 10) -> output", tuple(o.shape), " (no exception)")
print("_params['src'] :", aug._params["src"][0].tolist(), " -- indices 8 and 9 are outside a 6 x 8 input")
print("input  row 0 (8 values) :", [round(v, 6) for v in x[0, 0, 0].tolist()])
print("output row 0 (10 values):", [round(v, 6) for v in o[0, 0, 0].tolist()])
```

## Observed

```
kornia 0.9.0rc1 torch 2.14.0
input (1, 1, 6, 8) -> requested crop (10, 10) -> output (1, 1, 10, 10)  (no exception)
_params['src'] : [[0.0, 0.0], [9.0, 0.0], [9.0, 9.0], [0.0, 9.0]]  -- indices 8 and 9 are outside a 6 x 8 input
input  row 0 (8 values) : [0.0, 0.020833, 0.041667, 0.0625, 0.083333, 0.104167, 0.125, 0.145833]
output row 0 (10 values): [0.0, 0.014583, 0.03125, 0.047917, 0.064583, 0.08125, 0.097917, 0.114583, 0.13125, 0.145833]
```

The output row is a ten-point interpolation of the eight-point input row, with the same endpoints — the image was stretched, not cropped.

## Expected

Two coherent outcomes:

1. **Raise.** A crop larger than the input with padding explicitly disabled is a caller mistake; the message should name `size`, the input shape and `pad_if_needed`. This is the option that matches what `pad_if_needed` is for: the flag exists to opt into handling this case, so declining it should mean the case is not handled.
2. **Document the implicit resize** on `RandomCrop`, stating that an oversized `size` without padding upscales the input, and report it in the parameters so a caller can detect it.

The audit recommends (1). Either way it is a behavior change — today's silent upscale is what existing pipelines get — so it needs a release note rather than a quiet patch.

## Classification

**Semantic contract** — an out-of-range crop box is accepted and resampled instead of being rejected, so a crop silently becomes an upscale. Idiom: 4 (degenerate/singular input silently producing garbage — the `src` box exceeds the input's extent and is never checked against it).

## Disposition

`MERGE IN WINDOW — behavior change`. Raising where the library currently returns a tensor breaks any caller that relies on the present behavior, deliberately or not. Batch 6 documents the upscale on `RandomCrop` and pins the executed row above; it does not change it.

## Environment

kornia 0.9.0rc1 @ `d6a2c138` (main), torch 2.14.0, python 3.11, cpu. Found during the augmentation conventions audit (batch 6, tracking issue #4407); the batch documents the behavior above as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
